Reword some regex safety warnings (#54621)
dotnet/docs: includes/regex.md

> [!WARNING]

> > Unrestricted use of <xref:System.Text.RegularExpressions> with untrusted input can subject applications to [denial-of-service attacks](https://owasp.org/www-community/attacks/Regular_expression_Denial_of_Service_-_ReDoS). Consult [Best practices for regular expressions in .NET](/dotnet/standard/base-types/best-practices-regex) for guidance on how to safely use .NET regular expressions with untrusted input.

Author: I didn't change the existing formatting of this file.

> > Unrestricted use of <xref:System.Text.RegularExpressions> with untrusted input can subject applications to [denial-of-service attacks](https://owasp.org/www-community/attacks/Regular_expression_Denial_of_Service_-_ReDoS). Consult [Best practices for regular expressions in .NET](/dotnet/standard/base-types/best-practices-regex) for guidance on how to safely use .NET regular expressions with untrusted input.

Reviewer: ```suggestion
> Unrestricted use of <xref:System.Text.RegularExpressions> with untrusted input can subject applications to [denial-of-service attacks](https://owasp.org/www-community/attacks/Regular_expression_Denial_of_Service_-_ReDoS). Consult [Best practices for regular expressions in .NET](/dotnet/standard/base-types/best-practices-regex) for guidance on how to safely use .NET regular expressions with untrusted input.

```

> > Unrestricted use of <xref:System.Text.RegularExpressions> with untrusted input can subject applications to [denial-of-service attacks](https://owasp.org/www-community/attacks/Regular_expression_Denial_of_Service_-_ReDoS). Consult [Best practices for regular expressions in .NET](/dotnet/standard/base-types/best-practices-regex) for guidance on how to safely use .NET regular expressions with untrusted input.

Reviewer 2: ```suggestion
> Unrestricted use of <xref:System.Text.RegularExpressions> with untrusted input can subject applications to [denial-of-service attacks](https://owasp.org/www-community/attacks/Regular_expression_Denial_of_Service_-_ReDoS). Consult [Best practices for regular expressions in .NET](../docs/standard/base-types/best-practices-regex.md) for guidance on how to safely use .NET regular expressions with untrusted input.
```

> > Unrestricted use of <xref:System.Text.RegularExpressions> with untrusted input can subject applications to [denial-of-service attacks](https://owasp.org/www-community/attacks/Regular_expression_Denial_of_Service_-_ReDoS). Consult [Best practices for regular expressions in .NET](../docs/standard/base-types/best-practices-regex.md) for guidance on how to safely use .NET regular expressions with untrusted input.

Reviewer 2: ```suggestion
> Unrestricted use of <xref:System.Text.RegularExpressions> with untrusted input can subject applications to [denial-of-service attacks](https://owasp.org/www-community/attacks/Regular_expression_Denial_of_Service_-_ReDoS). Consult [Best practices for regular expressions in .NET](../docs/standard/base-types/best-practices-regex.md) for guidance on how to safely use .NET regular expressions with untrusted input.

```
